src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -371,6 +371,20 @@ void AwtWindow::RepositionSecurityWarning(JNIEnv *env)
 MsgRouting AwtWindow::WmWindowPosChanged(LPARAM windowPos) {
     WINDOWPOS * wp = (WINDOWPOS *)windowPos;
 
+    // There's no good way to detect partial maximization (e.g. Aero Snap),
```

Comment:
Does the bug affect all windows, or only frames/dialog? Can we move this code to the AwtFrame#WmWindowPosChanging?

Reply:
I believe it should only affect frames and dialogs, but I don't understand, why it wasn't done for `CheckWindowDPIChange` in the first place, AFAIU this hack is only relevant for frames and dialogs as well. I can move this code into AwtFrame, but then I'd need to loosen visibility of `prevScaleRec` and `m_screenNum`, I personally don't think it worth it. What do you think?